backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java

```diff
@@ -2,58 +2,199 @@
+        String role = parseString(r);
+        if (role == null) {
+            return PersonRole.UNKNOWN;
+        } 
+        try {
+            return PersonRole.valueOf(role.toUpperCase());
+        } catch (IllegalArgumentException e){
+            throw new IllegalGraphqlArgumentException("\""+r+"\" is not a valid role.");
+        }
+    }
+
+    public static String parseEmail(String e) {
+        String email = parseString(e);
+        if (email == null) {
+            return null;
+        } 
+        if (email.contains("@")) {
+            return email;
+        }
+        throw new IllegalGraphqlArgumentException("\""+e+"\" is not a valid email.");
+    }
+
+    private static final Set<String> RACES = Set.of(
+        "native", "asian", "black", "pacific", "white", "unknown", "refused"
+    );
+
+    public static String parseRace(String r) {
+        String race = parseString(r);
+        if (race == null) {
+            return null;
+        }
+        race = race.toLowerCase();
+        if (RACES.contains(race)) {
+            return race;
+        }
+        throw new IllegalGraphqlArgumentException("\""+r+"\" must be one of [" + String.join(", ", RACES) + "].");
+    }
+
+    private static final Set<String> ETHNICITIES = Set.of(
+        "hispanic", "not_hispanic"
+    );
+
+    public static String parseEthnicity(String e) {
+        String ethnicity = parseString(e);
+        if (ethnicity == null) {
+            return null;
+        }
+        ethnicity = ethnicity.toLowerCase();
+        if (ETHNICITIES.contains(ethnicity)) {
+            return ethnicity;
+        }
+        throw new IllegalGraphqlArgumentException("\""+e+"\" must be one of [" + String.join(", ", ETHNICITIES) + "].");
+    }
+
+    private static final Set<String> GENDERS = Set.of(
+        "male", "female", "other"
+    );
+
+    public static String parseGender(String g) {
+        String gender = parseString(g);
+        if (gender == null) {
+            return null;
+        }
+        gender = gender.toLowerCase();
+        if (GENDERS.contains(gender)) {
+            return gender;
+        }
+        throw new IllegalGraphqlArgumentException("\""+g+"\" must be one of [" + String.join(", ", GENDERS) + "].");
+    }
+
+    private static final Map<String, Boolean> YES_NO = Map.of(
+        "y", true,
+        "yes", true,
+        "n", false,
+        "no", false
+    );
+
+    public static Boolean parseYesNo(String v) {
+        String stringValue = parseString(v);
+        if (stringValue == null) {
+            return null;
+        }
+        Boolean boolValue = YES_NO.get(stringValue.toLowerCase());
+        if (boolValue == null) {
+            throw new IllegalGraphqlArgumentException("\""+v+"\" is not a valid value.");
+        }
+        return boolValue;
+    }
+
+    private static final Set<String> STATE_CODES = Set.of(
+        "AK", "AL", "AR", "AS", "AZ", "CA", "CO", "CT", "DC", "DE", "FL", "GA", "GU", "HI", "IA",
```

Comment:
Will this just be state codes, or should we be accepting territory codes as well such as PR for Puerto Rico or VI for Virgin Islands? The front-end doesn't currently have them either so they won't occur, but just noting we'll have to keep the two in sync.

Reply:
Yes we should eventually add territories. All of the address handling is a temporary solution  until we add the Smart Streets integration